backend/tests/test_engine_plugin_resource_manager.cpp

```
@@ -1,42 +1,715 @@
+{
+    const int num_attempts = 20;
+
+    for(int i = 0; i < num_attempts; ++i)
+    {
+        try
+        {
+            auto rm = create_resource_manager();
+            EXPECT_NE(rm, nullptr);
+
+            EXPECT_NO_THROW(rm->set_stream(nullptr));
+        }
+        catch(const std::exception& e)
+        {
+            SUCCEED() << "Construction failed with exception: " << e.what();
+        }
+    }
+}
+
+TEST_F(Engine_plugin_resource_manager_stress_test, ConcurrentAccess_MultipleThreadsCreatingManagers)
+{
+    const int num_threads = 4;
+    std::vector<std::thread> threads;
+    std::vector<std::vector<std::shared_ptr<Engine_plugin_resource_manager>>> all_managers(
+        static_cast<size_t>(num_threads));
+    std::atomic<int> successful_creations{0};
+
+    threads.reserve(static_cast<size_t>(num_threads));
+    for(int t = 0; t < num_threads; ++t)
+    {
+        threads.emplace_back([t, &all_managers, &successful_creations]() {
+            constexpr int managers_per_thread = 10;
+            for(int i = 0; i < managers_per_thread; ++i)
+            {
+                try
+                {
+                    auto rm = create_resource_manager();
+                    if(rm != nullptr)
+                    {
+                        all_managers[static_cast<size_t>(t)].push_back(rm);
+                        successful_creations++;
+
+                        rm->set_stream(nullptr);
+
+                        std::this_thread::sleep_for(std::chrono::microseconds(1));
+                    }
+                }
+                catch(const std::exception& e)
+                {
+                }
+            }
+        });
+    }
+
+    for(auto& thread : threads)
+    {
+        thread.join();
+    }
+
+    EXPECT_GT(successful_creations.load(), 0);
+
+    for(const auto& thread_managers : all_managers)
+    {
+        for(const auto& rm : thread_managers)
+        {
+            EXPECT_NE(rm, nullptr);
+        }
+    }
+}
```

I think it's good to have this threaded test just to verify it works properly in a threaded context.

The exceptions shouldn't be ignored, and should be fails. 
Since it's threaded I am not sure the best way to propagate the errors, but we shouldn't be catching and ignoring exceptions. No failures should occur in this case.